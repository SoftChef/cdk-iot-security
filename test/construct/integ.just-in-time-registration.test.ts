import * as path from 'path';
import '@aws-cdk/assert/jest';
import { SynthUtils } from '@aws-cdk/assert';
import {
  Function,
  InlineCode,
  Runtime,
} from '@aws-cdk/aws-lambda';
import { Bucket } from '@aws-cdk/aws-s3';
import { App, Stack } from '@aws-cdk/core';
import {
  CaRegistrator,
  DeviceActivator,
  JustInTimeRegistration,
  JustInTimeProvision,
  VerifiersFetcher,
  ReviewReceptor,
  ReviewAcceptionRole,
  JitrTopicRule,
  RegistrationConfigRole,
  DeviceCertificateGenerator,
} from '../../src/index';

describe('Test index.ts importation', () => {
  test('Create Functions', () => {
    process.env.BASE_PATH = __dirname;
    process.env.APPS_PATH = path.resolve(__dirname, '..', '..', 'src', 'lambda-assets');
    const app = new App();
    const stack = new Stack(app, 'test-stack');
    const anotherStack = new Stack(app, 'another-stack');
    const bucket = new Bucket(anotherStack, 'userProvidedBucket');
    new DeviceActivator(stack, 'testDeviceActivator');
    new VerifiersFetcher(stack, 'testVerifiersFetcher');
    new CaRegistrator(stack, 'testCaRegistrationFunction', {
      vault: {
        bucket: bucket,
        prefix: 'test',
      },
    });
    new DeviceCertificateGenerator(stack, 'testDeviceCertificateGenerator', {
      vault: {
        bucket: bucket,
      },
    });
    const reviewReceptor = new ReviewReceptor(stack, 'testReviewReceptor');
    new ReviewAcceptionRole(reviewReceptor, 'testReviewAcceptionRole', 'iot.amazonaws.com');
    new JitrTopicRule(reviewReceptor, 'testJitrTopicRule');
    new RegistrationConfigRole(stack, 'testJitpRole');
  });
});

describe('Test JustInTimeRegistration', () => {
  test('integration test', () => {
    process.env.BASE_PATH = __dirname;
    process.env.APPS_PATH = path.resolve(__dirname, '..', '..', 'src', 'lambda-assets');
    const app = new App();
    const stack = new Stack(app, 'test-stack');
    const verifierStack = new Stack(app, 'verifier-stack');
    const name = 'test-case';
    const anotherStack = new Stack(app, 'another-stack');
    const bucket = new Bucket(anotherStack, 'userProvidedBucket');
    new JustInTimeRegistration(stack, name, {
      verifiers: [
        new Function(verifierStack, name, {
          code: new InlineCode('exports.handler = () => { return true; }'),
          runtime: Runtime.NODEJS_12_X,
          handler: 'index.js',
        }),
      ],
      vault: {
        bucket: bucket,
        prefix: 'test',
      },
    });

    expect(SynthUtils.synthesize(stack).template).toMatchSnapshot();
    expect(stack).toCountResources('AWS::Lambda::Function', 3);
    expect(stack).toCountResources('AWS::IAM::Role', 4);
    expect(stack).toCountResources('AWS::SQS::Queue', 1);
  });

  test('integration test without providing verifier and prefix', () => {
    process.env.BASE_PATH = __dirname;
    process.env.APPS_PATH = path.resolve(__dirname, '..', '..', 'src', 'lambda-assets');
    const app = new App();
    const stack = new Stack(app, 'test-stack');
    const name = 'test-case';
    const anotherStack = new Stack(app, 'another-stack');
    const bucket = new Bucket(anotherStack, 'userProvidedBucket');
    new JustInTimeRegistration(stack, name, {
      vault: {
        bucket: bucket,
      },
    });
    expect(SynthUtils.synthesize(stack).template).toMatchSnapshot();
    expect(stack).toCountResources('AWS::Lambda::Function', 3);
    expect(stack).toCountResources('AWS::IAM::Role', 4);
    expect(stack).toCountResources('AWS::SQS::Queue', 1);
  });
});

describe('Test JustInTimeProvision', () => {
  test('Integration test', () => {
    process.env.BASE_PATH = __dirname;
    process.env.APPS_PATH = path.resolve(__dirname, '..', '..', 'src', 'lambda-assets');
    const app = new App();
    const stack = new Stack(app, 'test-stack');
    const name = 'test-case';
    const anotherStack = new Stack(app, 'another-stack');
    const bucket = new Bucket(anotherStack, 'userProvidedBucket');
    new JustInTimeProvision(stack, name, {
      vault: {
        bucket: bucket,
        prefix: 'test',
      },
    });
    expect(SynthUtils.synthesize(stack).template).toMatchSnapshot();
    expect(stack).toCountResources('AWS::Lambda::Function', 3);
    expect(stack).toCountResources('AWS::IAM::Role', 4);
  });
});