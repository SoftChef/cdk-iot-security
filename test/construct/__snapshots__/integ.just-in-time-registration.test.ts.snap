// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration test JustInTimeRegistration 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters3a15b45b155b1947f728bd801d2a9b93eab10a66ed3b8c1e281b4d61e28d89beArtifactHashD7E4E753": Object {
      "Description": "Artifact hash for asset \\"3a15b45b155b1947f728bd801d2a9b93eab10a66ed3b8c1e281b4d61e28d89be\\"",
      "Type": "String",
    },
    "AssetParameters3a15b45b155b1947f728bd801d2a9b93eab10a66ed3b8c1e281b4d61e28d89beS3Bucket4FEC42B8": Object {
      "Description": "S3 bucket for asset \\"3a15b45b155b1947f728bd801d2a9b93eab10a66ed3b8c1e281b4d61e28d89be\\"",
      "Type": "String",
    },
    "AssetParameters3a15b45b155b1947f728bd801d2a9b93eab10a66ed3b8c1e281b4d61e28d89beS3VersionKeyC3EBEEEE": Object {
      "Description": "S3 key for asset version \\"3a15b45b155b1947f728bd801d2a9b93eab10a66ed3b8c1e281b4d61e28d89be\\"",
      "Type": "String",
    },
    "AssetParameters7914903544f15f9f3ecb8300584f8c4196e7e0aa20dc6a7cbe9ac3b5c7eaa63dArtifactHash7BCF7240": Object {
      "Description": "Artifact hash for asset \\"7914903544f15f9f3ecb8300584f8c4196e7e0aa20dc6a7cbe9ac3b5c7eaa63d\\"",
      "Type": "String",
    },
    "AssetParameters7914903544f15f9f3ecb8300584f8c4196e7e0aa20dc6a7cbe9ac3b5c7eaa63dS3Bucket2362C509": Object {
      "Description": "S3 bucket for asset \\"7914903544f15f9f3ecb8300584f8c4196e7e0aa20dc6a7cbe9ac3b5c7eaa63d\\"",
      "Type": "String",
    },
    "AssetParameters7914903544f15f9f3ecb8300584f8c4196e7e0aa20dc6a7cbe9ac3b5c7eaa63dS3VersionKeyA0465D7F": Object {
      "Description": "S3 key for asset version \\"7914903544f15f9f3ecb8300584f8c4196e7e0aa20dc6a7cbe9ac3b5c7eaa63d\\"",
      "Type": "String",
    },
    "AssetParametersea69fa007256800e61fc21ea5d8a8c7240c2247b4f58b54ab22269e7883e7546ArtifactHashFEFB43A9": Object {
      "Description": "Artifact hash for asset \\"ea69fa007256800e61fc21ea5d8a8c7240c2247b4f58b54ab22269e7883e7546\\"",
      "Type": "String",
    },
    "AssetParametersea69fa007256800e61fc21ea5d8a8c7240c2247b4f58b54ab22269e7883e7546S3BucketE2D7FF12": Object {
      "Description": "S3 bucket for asset \\"ea69fa007256800e61fc21ea5d8a8c7240c2247b4f58b54ab22269e7883e7546\\"",
      "Type": "String",
    },
    "AssetParametersea69fa007256800e61fc21ea5d8a8c7240c2247b4f58b54ab22269e7883e7546S3VersionKey19247B31": Object {
      "Description": "S3 key for asset version \\"ea69fa007256800e61fc21ea5d8a8c7240c2247b4f58b54ab22269e7883e7546\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "JustInTimeRegistrationtestcaseCaRegistratortestcase40DBE3A2": Object {
      "DependsOn": Array [
        "JustInTimeRegistrationtestcaseCaRegistratortestcaseServiceRoleDefaultPolicy816F422E",
        "JustInTimeRegistrationtestcaseCaRegistratortestcaseServiceRole8F8171C3",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters7914903544f15f9f3ecb8300584f8c4196e7e0aa20dc6a7cbe9ac3b5c7eaa63dS3Bucket2362C509",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters7914903544f15f9f3ecb8300584f8c4196e7e0aa20dc6a7cbe9ac3b5c7eaa63dS3VersionKeyA0465D7F",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters7914903544f15f9f3ecb8300584f8c4196e7e0aa20dc6a7cbe9ac3b5c7eaa63dS3VersionKeyA0465D7F",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "BUCKET_NAME": Object {
              "Fn::ImportValue": "another-stack:ExportsOutputRefuserProvidedBucket2349EF2D343936C6",
            },
            "BUCKET_PREFIX": "test",
            "VERIFIERS": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "[\\"",
                  Object {
                    "Fn::ImportValue": "verifier-stack:ExportsOutputReftestcase9BC7DE4DD4B50F74",
                  },
                  "\\"]",
                ],
              ],
            },
          },
        },
        "Handler": "index.handler",
        "MemorySize": 256,
        "Role": Object {
          "Fn::GetAtt": Array [
            "JustInTimeRegistrationtestcaseCaRegistratortestcaseServiceRole8F8171C3",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 10,
      },
      "Type": "AWS::Lambda::Function",
    },
    "JustInTimeRegistrationtestcaseCaRegistratortestcaseDE8814EE": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "iam:PassRole",
                "iot:RegisterCACertificate",
                "iot:GetRegistrationCode",
                "iot:CreateTopicRule",
                "iot:TagResource",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "JustInTimeRegistrationtestcaseCaRegistratortestcaseDE8814EE",
        "Roles": Array [
          Object {
            "Ref": "JustInTimeRegistrationtestcaseCaRegistratortestcaseServiceRole8F8171C3",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "JustInTimeRegistrationtestcaseCaRegistratortestcaseServiceRole8F8171C3": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "JustInTimeRegistrationtestcaseCaRegistratortestcaseServiceRoleDefaultPolicy816F422E": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::ImportValue": "another-stack:ExportsOutputFnGetAttuserProvidedBucket2349EF2DArnA89CCA27",
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::ImportValue": "another-stack:ExportsOutputFnGetAttuserProvidedBucket2349EF2DArnA89CCA27",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "JustInTimeRegistrationtestcaseCaRegistratortestcaseServiceRoleDefaultPolicy816F422E",
        "Roles": Array [
          Object {
            "Ref": "JustInTimeRegistrationtestcaseCaRegistratortestcaseServiceRole8F8171C3",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "JustInTimeRegistrationtestcaseDeviceActivatortestcaseBE99699A": Object {
      "DependsOn": Array [
        "JustInTimeRegistrationtestcaseDeviceActivatortestcaseServiceRoleDefaultPolicyD0151B2D",
        "JustInTimeRegistrationtestcaseDeviceActivatortestcaseServiceRoleCD94255D",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersea69fa007256800e61fc21ea5d8a8c7240c2247b4f58b54ab22269e7883e7546S3BucketE2D7FF12",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersea69fa007256800e61fc21ea5d8a8c7240c2247b4f58b54ab22269e7883e7546S3VersionKey19247B31",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersea69fa007256800e61fc21ea5d8a8c7240c2247b4f58b54ab22269e7883e7546S3VersionKey19247B31",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "JustInTimeRegistrationtestcaseDeviceActivatortestcaseServiceRoleCD94255D",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "JustInTimeRegistrationtestcaseDeviceActivatortestcasePolicyDeviceActivatortestcaseE698297C": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "iot:DescribeCertificate",
                "iot:DescribeCACertificate",
                "iot:ListTagsForResource",
                "iot:CreateThing",
                "iot:CreatePolicy",
                "iot:AttachPolicy",
                "iot:AttachThingPrincipal",
                "iot:UpdateCertificate",
                "lambda:InvokeFunction",
                "lambda:InvokeAsync",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "JustInTimeRegistrationtestcaseDeviceActivatortestcasePolicyDeviceActivatortestcaseE698297C",
        "Roles": Array [
          Object {
            "Ref": "JustInTimeRegistrationtestcaseDeviceActivatortestcaseServiceRoleCD94255D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "JustInTimeRegistrationtestcaseDeviceActivatortestcaseServiceRoleCD94255D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "JustInTimeRegistrationtestcaseDeviceActivatortestcaseServiceRoleDefaultPolicyD0151B2D": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "sqs:ReceiveMessage",
                "sqs:ChangeMessageVisibility",
                "sqs:GetQueueUrl",
                "sqs:DeleteMessage",
                "sqs:GetQueueAttributes",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "JustInTimeRegistrationtestcaseReviewReceptortestcaseAB8D5DFE",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "JustInTimeRegistrationtestcaseDeviceActivatortestcaseServiceRoleDefaultPolicyD0151B2D",
        "Roles": Array [
          Object {
            "Ref": "JustInTimeRegistrationtestcaseDeviceActivatortestcaseServiceRoleCD94255D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "JustInTimeRegistrationtestcaseDeviceActivatortestcaseSqsEventSourceteststackJustInTimeRegistrationtestcaseReviewReceptortestcase3523F08BEF2BA3A9": Object {
      "Properties": Object {
        "BatchSize": 1,
        "EventSourceArn": Object {
          "Fn::GetAtt": Array [
            "JustInTimeRegistrationtestcaseReviewReceptortestcaseAB8D5DFE",
            "Arn",
          ],
        },
        "FunctionName": Object {
          "Ref": "JustInTimeRegistrationtestcaseDeviceActivatortestcaseBE99699A",
        },
      },
      "Type": "AWS::Lambda::EventSourceMapping",
    },
    "JustInTimeRegistrationtestcaseReviewReceptortestcaseAB8D5DFE": Object {
      "DeletionPolicy": "Delete",
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "JustInTimeRegistrationtestcaseReviewReceptortestcaseReviewAcceptionRoletestcaseA3AF3CD4": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "iot.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "sqs:SendMessageBatch",
                    "sqs:SendMessage",
                  ],
                  "Effect": "Allow",
                  "Resource": Object {
                    "Fn::GetAtt": Array [
                      "JustInTimeRegistrationtestcaseReviewReceptortestcaseAB8D5DFE",
                      "Arn",
                    ],
                  },
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "SqsPushPolicy",
          },
        ],
        "RoleName": "ReviewAcceptionRoleName-test-case",
      },
      "Type": "AWS::IAM::Role",
    },
    "JustInTimeRegistrationtestcaseReviewReceptortestcaseTopicRuletestcaseB619B96E": Object {
      "Properties": Object {
        "TopicRulePayload": Object {
          "Actions": Array [
            Object {
              "Sqs": Object {
                "QueueUrl": Object {
                  "Ref": "JustInTimeRegistrationtestcaseReviewReceptortestcaseAB8D5DFE",
                },
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "JustInTimeRegistrationtestcaseReviewReceptortestcaseReviewAcceptionRoletestcaseA3AF3CD4",
                    "Arn",
                  ],
                },
              },
            },
          ],
          "Sql": "SELECT * FROM '$aws/events/certificates/registered/#'",
        },
      },
      "Type": "AWS::IoT::TopicRule",
    },
    "JustInTimeRegistrationtestcaseVerifiersFetchertestcase4F2CFBBA": Object {
      "DependsOn": Array [
        "JustInTimeRegistrationtestcaseVerifiersFetchertestcaseServiceRoleFE2AC6E2",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters3a15b45b155b1947f728bd801d2a9b93eab10a66ed3b8c1e281b4d61e28d89beS3Bucket4FEC42B8",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters3a15b45b155b1947f728bd801d2a9b93eab10a66ed3b8c1e281b4d61e28d89beS3VersionKeyC3EBEEEE",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters3a15b45b155b1947f728bd801d2a9b93eab10a66ed3b8c1e281b4d61e28d89beS3VersionKeyC3EBEEEE",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "VERIFIERS": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "[\\"",
                  Object {
                    "Fn::ImportValue": "verifier-stack:ExportsOutputReftestcase9BC7DE4DD4B50F74",
                  },
                  "\\"]",
                ],
              ],
            },
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "JustInTimeRegistrationtestcaseVerifiersFetchertestcaseServiceRoleFE2AC6E2",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "JustInTimeRegistrationtestcaseVerifiersFetchertestcaseServiceRoleFE2AC6E2": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
